app/src/test/java/ar/scacchipa/jsonplaceholderviewer/PostRespositoryUnitTest.kt

```diff
@@ -0,0 +1,84 @@
+package ar.scacchipa.jsonplaceholderviewer
+
+import ar.scacchipa.jsonplaceholderviewer.data.*
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Test
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.dsl.module
+import org.koin.test.KoinTest
+import org.koin.test.inject
+import retrofit2.Response
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+
+class MockSourceData: ITypicodeSourceData {
+    override suspend fun getPosts(): Response<List<Post>> {
+        return Response.success( getMockPosts() )
+    }
+
+    override suspend fun getComment(id: String): Response<List<Comment>> {
+        return if (id.toInt() == 5) {
+            Response.success( getMockComments() )
+        } else {
+            Response.success( listOf() )
+        }
+    }
+}
+
+class PostRespositoryUnitTest: KoinTest {
+
+    private val postRepository: IPostRepository by inject()
+
+    @Test
+    fun shouldGetNinePost() = runBlocking {
```

Comment:
after reading some documentation, seems like runblocking is. not anymore the recommended way to run coroutines in unit tests. The google way since 2021 is to use a coroutine builder called runTest {}. To use it you need to add a depencency. I put you the link from jetbrains docuentations official, and also a handy post I found explaining why we should use from now on. I try it myself on your branch. using this new runTest builder and works like a charm.
blog post : https://craigrussell.io/2021/12/testing-android-coroutines-using-runtest/
jetbrain offical doc : https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/run-test.html